src/qz/utils/SystemUtilities.java

```diff
@@ -468,26 +469,43 @@ public static boolean setSystemLookAndFeel() {
      * @return <code>true</code> if the operation is successful
      */
     public static void centerDialog(Dialog dialog, Point position) {
-        if (position == null || position.getX() == 0 || position.getY() == 0) {
-            log.debug("Invalid dialog position provided: {}, we'll center on first monitor instead", position);
+        if (position == null) {
+            log.debug("No dialog position provided, we'll center on the primary monitor instead");
             dialog.setLocationRelativeTo(null);
             return;
         }
 
         //adjust for dpi scaling
         double dpiScale = getWindowScaleFactor();
         if (dpiScale == 0) {
-            log.debug("Invalid window scale value: {}, we'll center on first monitor instead", dpiScale);
+            log.debug("Invalid window scale value: {}, we'll center on the primary monitor instead", dpiScale);
             dialog.setLocationRelativeTo(null);
             return;
         }
 
-        Point p = new Point((int)(position.getX() * dpiScale), (int)(position.getY() * dpiScale));
-
-        //account for own size when centering
-        p.translate((int)(-dialog.getWidth() / 2.0), (int)(-dialog.getHeight() / 2.0));
+        Rectangle rect = new Rectangle((int)(position.x * dpiScale), (int)(position.y * dpiScale), dialog.getWidth(), dialog.getHeight());
+        rect.translate(-dialog.getWidth() / 2, -dialog.getHeight() / 2);
+        Point p = new Point((int)rect.getCenterX(), (int)rect.getCenterY());
         log.debug("Calculated dialog centered at: {}", p);
-        dialog.setLocation(p);
+
+        if (!isWindowLocationValid(rect)) {
+            log.debug("Dialog position provided is out of bounds: {}, we'll center on the primary monitor instead", p);
+            dialog.setLocationRelativeTo(null);
+            return;
+        }
+
+        dialog.setLocation(rect.getLocation());
+    }
+
```

Comment:
Forgot to add a doc for this function. I'll do that

Reply:
Consider keeping the point calculation, moving the window, then calculating the bounds with the rectangle logic.  The advantage of this approach is that we can write a generic on-screen-bounds calculation, e.g... 

```java
   /**
     * Validates if a given component is within screen bounds
     */
    public static boolean isComponentOnScreen(Component c) {
        // NOT YET TESTED
        Rectangle bounds = new Rectangle(c.getX(), c.getY(), c.getWidth(), c.getHeight());

        GraphicsDevice[] devices = GraphicsEnvironment.getLocalGraphicsEnvironment().getScreenDevices();
        Area area = new Area();
        for (GraphicsDevice gd : devices) {
            for (GraphicsConfiguration gc :  gd.getConfigurations()) {
                area.add(new Area(gc.getBounds()));
            }
        }
        return area.contains(bounds);
    }
```

Reply:
getLocationOnScreen
Throws:
    IllegalComponentStateException - if the component is not showing on the screen

looks like we may need to find another way

Reply:
Hmm.... Perhaps just `getLocation()` then?

Reply:
Since we don't move the component until AFTER we calculate its location, it's probably best to leave it as-is for now.